lib/deadpool-cyclone/src/instance/cyclone/local_uds.rs

```diff
@@ -486,6 +476,243 @@ fn temp_path_and_socket_from(
+                    .arg(sock)
+                    .arg("--decryption-key")
+                    .arg(&spec.cyclone_decryption_key_path)
+                    .arg("--lang-server")
+                    .arg(&spec.lang_server_cmd_path)
+                    .arg("--enable-watch");
+                if let Some(limit_requests) = spec.limit_requests {
+                    cmd.arg("--limit-requests").arg(limit_requests.to_string());
+                }
+                if let Some(timeout) = spec.watch_timeout {
+                    cmd.arg("--watch-timeout")
+                        .arg(timeout.as_secs().to_string());
+                }
+                if spec.ping {
+                    cmd.arg("--enable-ping");
+                }
+                if spec.resolver {
+                    cmd.arg("--enable-resolver");
+                }
+                if spec.action {
+                    cmd.arg("--enable-action-run");
+                }
+
+                Ok(LocalInstanceRuntime {
+                    runtime_strategy: LocalUdsRuntimeStrategy::LocalProcess,
+                    command: Some(cmd),
+                    child: None,
+                    container_id: None,
+                    docker_api: None,
+                })
+            }
+            LocalUdsRuntimeStrategy::LocalDocker => {
+                let mut cmd = vec![
+                    String::from("--bind-uds"),
+                    sock.to_string_lossy().to_string(),
+                    String::from("--decryption-key"),
+                    String::from("/tmp/key"),
+                    String::from("--lang-server"),
+                    String::from("/tmp/langserver"),
+                    String::from("--enable-watch"),
+                ];
+                if let Some(limit_requests) = spec.limit_requests {
+                    cmd.push(String::from("--limit-requests"));
+                    cmd.push(limit_requests.to_string())
+                }
+                if let Some(timeout) = spec.watch_timeout {
+                    cmd.push(String::from("--watch-timeout"));
+                    cmd.push(timeout.as_secs().to_string());
+                }
+                if spec.ping {
+                    cmd.push(String::from("--enable-ping"));
+                }
+                if spec.resolver {
+                    cmd.push(String::from("--enable-resolver"));
+                }
+                if spec.action {
+                    cmd.push(String::from("--enable-action-run"));
+                }
+
+                let docker = Docker::connect_with_local_defaults()?;
+                let rand_string: String = thread_rng()
+                    .sample_iter(&Alphanumeric)
+                    .take(10)
+                    .map(char::from)
+                    .collect();
+
+                let socket_dir = sock
+                    .parent()
+                    .expect("socket path not available")
+                    .display()
+                    .to_string();
```

Comment:
Going through the `Display` trait may not be ideal here. Perhaps [`to_str`](https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.to_str) would be the way to go?